platform/apns/service.go

```diff
@@ -121,13 +124,30 @@ func updateClient(svc *PushService, sub pubsub.Subscriber) error {
 	return nil
 }
 
+func newClient(cert tls.Certificate) (*http.Client, error) {
+	config := &tls.Config{
+		Certificates: []tls.Certificate{cert},
+	}
+	config.BuildNameToCertificate()
+	transport := &http.Transport{
+		TLSClientConfig: config,
+		IdleConnTimeout: 90 * time.Second,
+	}
+
+	if err := http2.ConfigureTransport(transport); err != nil {
+		return nil, err
+	}
+
+	return &http.Client{Transport: transport}, nil
```

Comment:
the `http.Client` field also has a timeout which covers a few other places in the request that could time. out. Lets set this to 
```
return &http.Client{
  Transport: transport,
  Timeout: 20 * time.Second,
}, nil
```

Reply:
Added in 021962d